src/hooks/achievement-rule-checker.js

```diff
@@ -24,9 +24,18 @@ module.exports = function (options = {}) {
         } else {
           achievementService.create({user_id: context.data.user_id, name: achievementRule.name, amount: 1});
         }
+
+        for (const replaceName of achievementRule.replaces) {
+          const replacedAchievement = await achievementService.find({
+            query: {
+              user_id: context.data.user_id,
+              name: replaceName
+            }
+          });
```

Comment:
`replacedAchievement` can be an empty array if the to-be-replaced achievement was not granted to the user before. The next line will break then.

Reply:
But in that case it wouldn't enter the for loop in the first place? `achievementRule.replaces` would be empty and therefore there wouldn't be a `replaceName` to begin with.